pkg/compactor/shard_aware_deduplicate_filter.go

```diff
@@ -31,9 +31,10 @@ func NewShardAwareDeduplicateFilter() *ShardAwareDeduplicateFilter {
 	return &ShardAwareDeduplicateFilter{}
 }
 
-// Filter filters out duplicate blocks that can be formed
-// from two or more overlapping blocks that fully submatches the source blocks of the older blocks.
+// Filter filters out from metas, the initial map of blocks, all the blocks that are contained in other, compacted, blocks from metas.
+// The removed blocks are source blocks of the blocks that remain in metas after the filtering is executed.
```

Comment:
I'm not sure I understand the new sentence.

Reply:
@pstibrany I am happy to rewrite it.
// Filter filters out duplicate blocks that can be formed
// from two or more overlapping blocks that fully submatches the source blocks of the older blocks.

My vision of the Filter method is that it removes from the map of blocks passed as input all those blocks for which there is another block in the map in which the first block is contained, i.e., for which the first block is a source block. Am I wrong?